### ISSUE TITLE
Display the logo of the activities within the graph 

### ISSUE THREAD

**Issue by bytebang** [bytebang/moodle-mod_activitymap#14]
**Is your feature request related to a problem? Please describe.**
It is not realted to a problem, but it would enhance understanding.

**Describe the solution you'd like**
The graph should display the logo for the activities in the headline (as small icon)

**Describe alternatives you've considered**
No alternatives

**Additional context**
No Context

**Comment by bytebang** [comment]
After a few hours of trying i figured out here https://github.com/mdaines/viz.js/wiki/API#images-and-files-options that all images which should be displayed from within a graph have to be preloaded into Emscriptens in-memory file system.

Since the activity map plugin is (in the moment) built in a way where the gvizdot.php (where the graph is constructed) does not speak to the view.php (where the graph is displayed within viz.js) it is currently not possible to display images within the graph.


In order to do this, i would have to remember all embedded images and inject them just before the graph is displayed

**Comment by bytebang** [comment]
If somebody likes to play / fiddle with it: Here is how i would display the logo within the graph

```
diff --git a/gvizdot.php b/gvizdot.php
index 3afe886..044afb0 100644
--- a/gvizdot.php
+++ b/gvizdot.php
@@ -377,7 +377,11 @@ foreach ($modinfo->cms as $id => $cm) {
         $gvnodeattributes["shape"] = $activitymap->elementshape;
         $gvnodeattributes["label"] = "<b>" . htmlentities($cm->name) . "</b>";
         $gvnodeattributes["tooltip"] = htmlentities($cm->name);
-
+      
+        // Get the icon url of the activity and append it
+        $cmIconUrl = new moodle_url('/mod/' . $cm->modname . '/pix/icon.png');
+        $gvnodeattributes["label"] = "<TABLE BORDER=\"0\" CELLPADDING=\"0\" CELLSPACING=\"0\"><TR><TD><IMG SRC=\"$cmIconUrl\"/></TD><TD>" . $gvnodeattributes["label"] . "</TD></TR></TABLE>";
+        
```

**Comment by bytebang** [comment]
solved in 7176ef9
